core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -348,17 +361,28 @@ export class ContributorStatsComponent {
   }
 
   goToNextPage(page: StatsPage): void {
+    if (
+      this.statsData[this.type].currentPageEndIndex >=
+      this.statsData[this.type].data.length) {
+      throw new Error('There are no more pages to go.');
+    }
     page.currentPageStartIndex += this.ITEMS_PER_PAGE;
     page.currentPageEndIndex += this.ITEMS_PER_PAGE;
   }
 
   goToPreviousPage(page: StatsPage): void {
+    if (
+      this.statsData[this.type].currentPageStartIndex === 0) {
+      throw new Error('There are no more pages to go back.');
+    }
     page.currentPageStartIndex -= this.ITEMS_PER_PAGE;
     page.currentPageEndIndex -= this.ITEMS_PER_PAGE;
   }
 
+  // This method gives the direction how the stats columns should be sorted.
   getColumnSortDirection(): number {
```

Comment:
I'm confused by this. There's only one direction and it's represented by a zero. That doesn't really make sense to me -- I would expect something like "+1" for normal and "-1" for reverse. But here there's only one value in the enum so I don't even understand why we need this function at all.

What is this actually representing? Is it actually a sort direction? If so what are the two values for "usual order" and "reverse order"?

Reply:
What I am doing is getting column names in the order I specified in COLUMNS object. According to the Angular documentation, the keyvalue pipe sorts the items by key order by default. We can provide a comparer function to change that order, and sort the items according to the key, to the value, or to the entry order of the properties in the object.

Since I do not want to sort according to the key or to the value I just added a comparer function to get the items in the order I had specified in the COLUMNS object. Do I still need to add the reverse or normal order as well(including ENUMS for them too)?

Reply:
Then this doesn't sound like a direction. getKeyToSortBy() perhaps?

But is the key always going to be 0? If so, then why do we need this function?

I think the important thing is to provide enough documentation in the code explaining what is going on. You don't need to add the reverse order if that is not an intended use case; the problem here is that I cannot understand what this code is doing and what the implication of returning 0 is.

Reply:
Basically what we need is a constant in order to preserve the original key value order.

I changed the function name as `provideOriginalOrder`

I access this as `*ngFor="let column of COLUMNS[type] | keyvalue: provideOriginalOrder"` in the template. We could have directly use *ngFor="let column of COLUMNS[type] | keyvalue: 0"` instead of creating a function to preserve the original order of keys. But it gives console errors and the fix was to return 0(any constant from a function). Source: https://stackoverflow.com/questions/52793944/angular-keyvalue-pipe-sort-properties-iterate-in-order